docs/source/integration-guide.rst

```diff
@@ -0,0 +1,189 @@
+   at runtime as environment variables. Scripts can make use of any of these environment variables but must
+   ensure that they are written in a way that prevents accidentally leaking Atlas API credentials and
+   MongoDB user credentials.
+
+
+.. note::
+
+   The installer script will be executed with the working directory set to the ``astrolabe`` project root.
+
+
+Wrapping native workload executors with a shell script
+------------------------------------------------------
+
+It might be difficult for some driver languages to create a single executable file for the workload executor as
+required by ``astrolabe``. In such cases, drivers may choose to wrap the actual call to a natively implemented
+workload executor in a shell script such that exposes the API desired by the
+:ref:`workload-executor-specification` specification.
+
+.. note::
+
+   PyMongo's ``astrolabe`` integration uses this pattern to implement its
+   `workload executor <https://github.com/mongodb-labs/drivers-atlas-testing/blob/master/.evergreen/python/pymongo/workload-executor.sh>`_.
+   Check it out for a practical example of what this looks like.
+
+Certain special considerations apply to drivers that choose to implement their workload executors in this manner:
+
+* The shell script used to wrap the native workload executor implementation MUST ensure that it traps the
+  ``INT`` signal and sets a zero exit-code to prevent the Test Orchestrator from misconstruing all interruptions
+  of the native workload executor as failures (shell scripts will set always set a non-zero exit-code when they
+  terminate after encountering the ``INT`` signal). To do this, add the following snippet to your script *before*
+  the call to your native workload executor::
+
+    trap "exit 0" INT
+
+* The shell script MUST NOT call the native workload executor as a background process (i.e. using ``&`` control
+  operator). Doing so will cause the native workload executor to run in the background in a subshell and make it
+  impossible for the Test Orchestrator to capture the STDOUT and STDERR streams of the workload executor.
+
+
+----------------------------------------------
+Step 2: Implementing a Driver Installer Script
+----------------------------------------------
+
+Drivers must implement this standalone script to perform all setup/installation-related tasks for their driver.
+The installer script MUST be saved in the ``DRIVER_DIRNAME`` directory under the name ``install-driver.sh``.
+The executable permission bit MUST be set for the install script file before it is committed to git.
+
+This script can be used to perform any number of arbitrary tasks related to setting up the environment for
+the workload executor to be executed within. It MUST NOT however, clone the driver source repository.
+
+.. note::
+
+   All Evergreen expansions for a given build variant are available to the installer script
```

Comment:
didn't we decide that this file itself could output expansion updates? I don't think I see that discussed here.

Reply:
We no longer need to write a file to do expansion updates. We use `subprocess.exec`(https://github.com/evergreen-ci/evergreen/wiki/Project-Commands#subprocessexec) with the `add_expansions_to_env` option so we don't need to mess around with the clunky expansions.yml files.

Reply:
so you're saying that any environment variable I set in my `install-driver.sh` script is automatically added to expansions that I can use in the next phase? If that's not the case, how does someone implementing this expose environment variables to use in the next phase if needed?

Reply:
Actually, you need to set the environment variables when setting up your driver and/or runtime in the evergreen configuration. I will make a note of it here so that it is clear. E.g. `PYTHON_BINARY` and `PYMONGO_VIRTUALENV_NAME` are variables that are need by pymongo's installer and workload executor scripts and you can see where we define them in `.evergreen/config.yml`.

Reply:
Could it be the case that some driver needs to dynamically set a variable? What if part of `install-driver.sh` for the Rust driver uses `rustup` to install a custom version of Rust, and then needs to refer to that location in the `workload-executor.sh` script?

Reply:
Yes, I suppose this could be needed. That being said, usually the tool like `rustup` will provide a CLI-based method to retrieve the path of what was installed, OR it would let you invoke your executable with what was downloaded. In the case of `rustup` for instance, if you want to run the test with rust nightly, you can do `rustup run nightly rustc foo.rs`.

To put it another way, I am not entirely convinced that we should add the scaffolding for expansion files unless absolutely necessary and at this point, I am not sure that we know it is necessary.